driver/src/contracts/snapp_contract.rs

```diff
@@ -109,163 +115,80 @@ impl SnappContract for SnappContractImpl {
     }
 
     fn get_current_state_root(&self) -> Result<H256> {
-        self.base
-            .contract
-            .query("getCurrentStateRoot", (), None, Options::default(), None)
-            .wait()
-            .map_err(DriverError::from)
+        Ok(self.instance.get_current_state_root().call().wait()?.into())
     }
 
     fn get_current_deposit_slot(&self) -> Result<U256> {
-        self.base
-            .contract
-            .query("getCurrentDepositIndex", (), None, Options::default(), None)
-            .wait()
-            .map_err(DriverError::from)
+        Ok(self.instance.get_current_deposit_index().call().wait()?)
     }
 
     fn get_current_withdraw_slot(&self) -> Result<U256> {
-        self.base
-            .contract
-            .query(
-                "getCurrentWithdrawIndex",
-                (),
-                None,
-                Options::default(),
-                None,
-            )
-            .wait()
-            .map_err(DriverError::from)
+        Ok(self.instance.get_current_withdraw_index().call().wait()?)
```

Comment:
This sequence of commands seems to be a common theme. Not that theres anything wrong with that.

Reply:
It is! Working as designed :stuck_out_tongue: